pipeline/data/download-mono.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+"""
+Downloads a monolingual dataset, shuffles it, and truncates it to a maximum amount of sentences.
+
+Kinds:
+   taskcluster/kinds/dataset/kind.yml
+
+Example usage:
+
+    pipeline/data/download-mono.py                  \\
+        --dataset news-crawl_news.2021              \\
+        --language en                               \\
+        --max_sentences 100000000                   \\
+        --artifacts $TASK_WORKDIR/artifacts
+
+Artifacts:
+
+    artifacts
+    └── news.2021.en.zst
+"""
+
+import argparse
+import gzip
+import io
+import os
+import re
+from random import Random
+from typing import Iterator, Optional
+
+import requests
+import zstandard
+from google.cloud.storage.fileio import BlobReader
+
+from pipeline.utils import Dataset, attempt_mocked_request
+from pipeline.utils.downloads import google_cloud_storage
+from pipeline.utils.logging import get_logger
+
+# TODO(CJK) - Issue #424
+MAX_WORDS_IN_SENTENCE = 100
+
+CURRENT_FOLDER = os.path.dirname(os.path.abspath(__file__))
+IMPORTERS_PATH = os.path.abspath(os.path.join(CURRENT_FOLDER, "mono"))
+
+logger = get_logger("download_mono")
+
+
+def shuffle_and_truncate(
+    line_stream: Iterator[str], dataset: Dataset, file_destination: str, max_sentences: int
+):
+    lines: list[str] = []
+
+    logger.info("Load the stream into memory, discarding sentences that are too long.")
+    for line in line_stream:
+        if len(line.split()) < MAX_WORDS_IN_SENTENCE:
```

Comment:
I'm not sure we should do this cleaning on this stage. Probably better to remove from here and leave it up to cleaning scripts.

Reply:
Do you want me to file a follow-up here?